test/quantization/test_numeric_suite.py

```diff
@@ -14,61 +16,93 @@ class EagerModeNumericSuiteTest(QuantizationTestCase):
     def test_compare_weights(self):
         r"""Compare the weights of float and quantized conv layer
         """
+
+        def compare_and_validate_results(float_model, q_model):
+            weight_dict = compare_weights(
+                float_model.state_dict(), q_model.state_dict()
+            )
+            self.assertEqual(len(weight_dict), 1)
+            for k, v in weight_dict.items():
+                self.assertTrue(v["float"].shape == v["quantized"].shape)
+
         # eager mode
         annotated_conv_model = AnnotatedConvModel().eval()
         quantized_annotated_conv_model = quantize(
             annotated_conv_model, default_eval_fn, self.img_data
         )
-        weight_dict = compare_weights(
-            annotated_conv_model.state_dict(),
-            quantized_annotated_conv_model.state_dict(),
+        compare_and_validate_results(
+            annotated_conv_model, quantized_annotated_conv_model
         )
-        self.assertEqual(len(weight_dict), 1)
-        for k, v in weight_dict.items():
-            self.assertTrue(v["float"].shape == v["quantized"].shape)
 
         annotated_conv_bn_relu_model = AnnotatedConvBnReLUModel().eval()
         annotated_conv_bn_relu_model.fuse_model()
         quantized_annotated_conv_bn_relu_model = quantize(
             annotated_conv_bn_relu_model, default_eval_fn, self.img_data
         )
-        weight_dict = compare_weights(
-            annotated_conv_bn_relu_model.state_dict(),
-            quantized_annotated_conv_bn_relu_model.state_dict(),
+        compare_and_validate_results(
+            annotated_conv_bn_relu_model, quantized_annotated_conv_bn_relu_model
         )
-        self.assertEqual(len(weight_dict), 1)
-        for k, v in weight_dict.items():
-            self.assertTrue(v["float"].shape == v["quantized"].shape)
 
     def test_compare_model_stub(self):
         r"""Compare the output of quantized conv layer and its float shadow module
         """
+
+        def compare_and_validate_results(float_model, q_model, module_swap_list, data):
+            ob_dict = compare_model_stub(float_model, q_model, module_swap_list, data)
+            self.assertEqual(len(ob_dict), 1)
+            for k, v in ob_dict.items():
+                self.assertTrue(v["float"].shape == v["quantized"].shape)
```

Comment:
In addition, can we check if the keys in the observer dict match what is expected? Are we obtaining the activations at all the points in the model?

Reply:
We will get the activations of the modules that are in the white list, may not be all the points in the model, but should cover most.